tests/glsl-intrinsic/intrinsic-functional.slang

```diff
@@ -0,0 +1,2319 @@
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=DX11):-allow-glsl -compute -dx11 -output-using-type
+//TEST(compute):COMPARE_COMPUTE_EX(filecheck-buffer=DX12):-allow-glsl -compute -dx12 -use-dxil -output-using-type -profile sm_6_0
+//TEST(compute, vulkan):COMPARE_COMPUTE_EX(filecheck-buffer=VK):-allow-glsl -compute -vk -emit-spirv-directly -output-using-type -profile sm_6_0 -render-feature hardware-device
+
+#define TRIG_FUNCTIONS 0
+#define EXP_FUNCTIONS 1
+#define COMMON_FUNCTIONS 2
+#define PACK_FUNCTIONS 3
+#define GEOMETRIC_FUNCTIONS 4
+#define MATRIX_FUNCTIONS 5
+#define VEC_FUNCTIONS 6
+#define INT_FUNCTIONS 7
+
+
+float AngleAndTrig()
```

Comment:
As described on the GLSL spec document, we should test all of types described as "genFType"
https://registry.khronos.org/OpenGL/specs/gl/GLSLangSpec.4.60.pdf

Please check "tests/glsl-intrinsic/intrinsic-basic.slang" for how different types are tested without copy and paste .
```
__generic<let N : int>
bool dimNTypeFuncs()
{
    typealias genFType = vector<float, N>;
    typealias genDType = vector<double, N>;
    typealias genIType = vector<int, N>;
    typealias genUType = vector<uint, N>;
    typealias genBType = vector<bool, N>;
    typealias vec      = vector<float, N>;
    typealias ivec     = vector<int, N>;
    typealias uvec     = vector<uint, N>;
    typealias bvec     = vector<bool, N>;

    genFType outGenFType;
    genDType outGenDType;
    genIType outGenIType, outGenIType2;
    genUType outGenUType, outGenUType2;
```